### PR TITLE
Add loc comments and fix Windows installer bug

### PR DESCRIPTION
**Pull request by dougbu**
- #44167, escape square brackets in `[directory]`
- add loc comments to lock WiX substitutions

### CODE REVIEW

File: src/Installers/Windows/SharedFrameworkBundle/thm.wxl [hunk 1]
```diff
@@ -1,15 +1,15 @@
 <?xml version="1.0" encoding="utf-8"?>
 <WixLocalization Culture="en-us" Language="1033" xmlns="http://schemas.microsoft.com/wix/2006/localization">
-  <String Id="Caption">[BundleNameFull] Setup</String>
-  <String Id="Title">[BundleNameShort]</String>
-  <String Id="SubTitle">[BundleNameSub]</String>
+  <String Id="Caption"><!--_locComment_text="{Locked='[BundleNameFull]'}"-->[BundleNameFull] Setup</String>
+  <String Id="Title"><!--_locComment_text="{Locked}"-->[BundleNameShort]</String>
+  <String Id="SubTitle"><!--_locComment_text="{Locked}"-->[BundleNameSub]</String>
```
Comment (dougbu) [inline src/Installers/Windows/SharedFrameworkBundle/thm.wxl:5]: @cristianosuzuki77 is this the correct application of your suggestion in [ticket 710644, [SW] Need for loc comments in .wxl files?](https://ceapex.visualstudio.com/CEINTL/_workitems/edit/710644/)❔

File: src/Installers/Windows/WindowsHostingBundle/thm.wxl [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
   <String Id="InstallHeader">Welcome</String>
-  <String Id="InstallMessage">Setup will install [WixBundleName] on your computer. Click install to continue, options to set the install directory or Close to exit.</String>
-  <String Id="InstallVersion">Version [WixBundleVersion]</String>
+  <String Id="InstallMessage"><!--_locComment_text="{Locked='[WixBundleName]'"-->Setup will install [WixBundleName] on your computer. Click install to continue, options to set the install directory or Close to exit.</String>
+  <String Id="InstallVersion"><!--_locComment_text="{Locked='[WixBundleVersion]'"-->Version [WixBundleVersion]</String>
   <String Id="ConfirmCancelMessage">Are you sure you want to cancel?</String>
   <String Id="ExecuteUpgradeRelatedBundleMessage">Previous version</String>
   <String Id="HelpHeader">Setup Help</String>
-  <String Id="HelpText">/install | /repair | /uninstall | /layout [directory] - installs, repairs, uninstalls or
+  <String Id="HelpText">/install | /repair | /uninstall | /layout [\[]directory[\]] - installs, repairs, uninstalls or
```
Comment (wtgodbe) [inline src/Installers/Windows/WindowsHostingBundle/thm.wxl:12]: Reviewing just this part as it appears to be the bugfix - seems reasonable

**Review by wtgodbe — APPROVED** [review]

**Comment by dougbu** [comment]
Internal test build is [#20220926.2](https://dev.azure.com/dnceng/internal/_build/results?buildId=2004502&view=results). That will uncover any problems in this update. CI here will match a rolling public build and provide no new information about the changes.

**Comment by dougbu** [comment]
/fyi build [#20220926.2](https://dev.azure.com/dnceng/internal/_build/results?buildId=2004502&view=results) succeeded. @cristianosuzuki77 is checking the created loc project&hellip;

File: src/Installers/Windows/WindowsHostingBundle/thm.wxl [hunk 3]
```diff
@@ -1,15 +1,15 @@
 <?xml version="1.0" encoding="utf-8"?>
 <WixLocalization Culture="en-us" Language="1033" xmlns="http://schemas.microsoft.com/wix/2006/localization">
-  <String Id="Caption">[WixBundleName] Setup</String>
-  <String Id="Title">[BundleNameShort]</String>
-  <String Id="SubTitle">[BundleNameSub]</String>
+  <String Id="Caption"><!--_locComment_text="{Locked='[WixBundleName]'"-->[WixBundleName] Setup</String>
```
Comment (crsuzukimsft) [inline src/Installers/Windows/WindowsHostingBundle/thm.wxl:3]: missing the closing }

File: src/Installers/Windows/WindowsHostingBundle/thm.wxl [hunk 4]
```diff
@@ -18,7 +18,7 @@
 /norestart   - suppress any attempts to restart. By default UI will prompt before restart.
 /log log.txt - logs to a specific file. By default a log file is created in %TEMP%.</String>
   <String Id="HelpCloseButton">&amp;Close</String>
-  <String Id="InstallLicenseLinkText">[WixBundleName] &lt;a href="#"&gt;license terms&lt;/a&gt;.</String>
+  <String Id="InstallLicenseLinkText"><!--_locComment_text="{Locked='[WixBundleName]'"-->[WixBundleName] &lt;a href="#"&gt;license terms&lt;/a&gt;.</String>
```
Comment (crsuzukimsft) [inline src/Installers/Windows/WindowsHostingBundle/thm.wxl:21]: missing the closing }

File: src/Installers/Windows/WindowsHostingBundle/thm.wxl [hunk 5]
```diff
@@ -60,10 +60,10 @@
   <String Id="FilesInUseCancelButton">&amp;Cancel</String>
 
   <!-- Custom UI strings -->
-  <String Id="Welcome">Welcome to the [WixBundleName] Setup.</String>
+  <String Id="Welcome"><!--_locComment_text="{Locked='[WixBundleName]'"-->Welcome to the [WixBundleName] Setup.</String>
```
Comment (crsuzukimsft) [inline src/Installers/Windows/WindowsHostingBundle/thm.wxl:63]: missing the closing }

File: src/Installers/Windows/WindowsHostingBundle/thm.wxl [hunk 6]
```diff
@@ -60,10 +60,10 @@
...
   <String Id="InstallResetIIS">Please restart IIS after the installation completes. You can find additional information &lt;a href="https://go.microsoft.com/fwlink/?LinkId=798277"&gt;here&lt;/a&gt;.</String>
   <String Id="InstallNoIIS">IIS is not enabled on this machine. If you intend to run ASP.NET Core applications with IIS, you must install IIS before running this installer. You can find additional information &lt;a href="https://go.microsoft.com/fwlink/?LinkId=798277"&gt;here&lt;/a&gt;.</String>
-  <String Id="EulaPrivacy">[WixBundleName] &lt;a href="https://go.microsoft.com/fwlink/?LinkId=329770"&gt;license terms&lt;/a&gt; and &lt;a href="https://go.microsoft.com/fwlink/?LinkId=786378"&gt;privacy statement&lt;/a&gt;.</String>
+  <String Id="EulaPrivacy"><!--_locComment_text="{Locked='[WixBundleName]'"-->[WixBundleName] &lt;a href="https://go.microsoft.com/fwlink/?LinkId=329770"&gt;license terms&lt;/a&gt; and &lt;a href="https://go.microsoft.com/fwlink/?LinkId=786378"&gt;privacy statement&lt;/a&gt;.</String>
```
Comment (crsuzukimsft) [inline src/Installers/Windows/WindowsHostingBundle/thm.wxl:66]: missing closing }

File: src/Installers/Windows/SharedFrameworkBundle/thm.wxl (hunk 1 above)
Comment (crsuzukimsft) [reply src/Installers/Windows/SharedFrameworkBundle/thm.wxl:5]: Yes, looks good.

File: src/Installers/Windows/WindowsHostingBundle/thm.wxl (hunk 3 above)
Comment (dougbu) [reply src/Installers/Windows/WindowsHostingBundle/thm.wxl:3]: Yeah, now I see that I did that a lot. Will fix **Real Soon Now:tm:**

**Comment by dougbu** [comment]
Anything else to address before I let the CI complete and merge this @cristianosuzuki77❔

**Comment by dougbu** [comment]
&hellip; for example, should I run the pipeline again❔

**Review by crsuzukimsft — APPROVED** [review]

**Pull request merged**